src/ray/raylet/task_dependency_manager.cc

```diff
@@ -80,22 +80,58 @@ std::vector<TaskID> TaskDependencyManager::HandleObjectLocal(
 
   // Find any tasks that are dependent on the newly available object.
   std::vector<TaskID> ready_task_ids;
+  std::vector<TaskID> task_ids_to_remove;
   auto creating_task_entry = required_tasks_.find(ComputeTaskId(object_id));
   if (creating_task_entry != required_tasks_.end()) {
     auto object_entry = creating_task_entry->second.find(object_id);
     if (object_entry != creating_task_entry->second.end()) {
       for (auto &dependent_task_id : object_entry->second) {
-        auto &task_entry = task_dependencies_[dependent_task_id];
-        task_entry.num_missing_dependencies--;
-        // If the dependent task now has all of its arguments ready, it's ready
-        // to run.
-        if (task_entry.num_missing_dependencies == 0) {
-          ready_task_ids.push_back(dependent_task_id);
+        auto it = task_dependencies_.find(dependent_task_id);
+        RAY_CHECK(it != task_dependencies_.end());
+        auto &task_entry = it->second;
+
+        // If the task called ray.get on the object, then the task may now be
+        // ready to run.
+        if (task_entry.get_dependencies.find(object_id) !=
+            task_entry.get_dependencies.end()) {
+          task_entry.num_missing_dependencies--;
+          // If the dependent task now has all of its arguments ready, it's
+          // ready to run.
+          if (task_entry.num_missing_dependencies == 0) {
+            ready_task_ids.push_back(dependent_task_id);
+          }
+        }
+
+        // If the task also called ray.wait on the object, then we can now
+        // remove the task's dependency on the object.
+        auto wait_it = task_entry.wait_dependencies.find(object_id);
+        if (wait_it != task_entry.wait_dependencies.end()) {
+          // The object is now local, so the ray.wait call has been fulfilled.
+          task_entry.wait_dependencies.erase(wait_it);
```

Comment:
Why don't we need an analogous call like `task_entry.get_dependencies.erase(wait_it);` in the block above?